src/Sylius/Component/Association/Model/Association.php

```
@@ -0,0 +1,135 @@
+     */
+    protected $owner;
+
+    /**
+     * @var Collection<Associatable>
+     */
+    protected $associatedObjects;
+
+    /**
+     * @var \DateTime
+     */
+    protected $createdAt;
+
+    /**
+     * @var \DateTime
+     */
+    protected $updatedAt;
+
+    public function __construct()
+    {
+        $this->createdAt = new \DateTime();
+        $this->updatedAt = new \DateTime();
+        $this->associatedObjects = new ArrayCollection();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getType()
+    {
+        return $this->type;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setType(AssociationTypeInterface $type)
+    {
+        $this->type = $type;
+    }
+
+    /**
+     * @return Associatable
+     */
+    public function getOwner()
+    {
+        return $this->owner;
+    }
+
+    /**
+     * @param Associatable $owner
+     */
+    public function setOwner(Associatable $owner = null)
+    {
+        $this->owner = $owner;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAssociatedObjects()
+    {
+        return $this->associatedObjects;
+    }
+
+    /**
+     * @param Collection $associatedObjects
+     */
+    public function setAssociatedObjects(Collection $associatedObjects)
+    {
+        $this->associatedObjects = $associatedObjects;
+    }
+
+    /**
+     * @param Associatable $associatedObject
+     *
+     * @return bool
+     */
+    public function hasAssociatedObject(Associatable $associatedObject)
+    {
+        return $this->associatedObjects->contains($associatedObject);
+    }
+
+    /**
+     * @param Associatable $associatedObject
+     */
+    public function addAssociatedObject(Associatable $associatedObject)
+    {
+        if (!$this->hasAssociatedObject($associatedObject)) {
+            $this->associatedObjects->add($associatedObject);
```

Same logic is in whole Product class. @pjedrzejewski should we throw an exception here?